include/mata/nfa-plumbing.hh

```diff
@@ -0,0 +1,153 @@
+/* nfa-plumbings.hh -- Wrapping up different supporting functions
+ *
+ * Copyright (c) 2022
+ *
+ * This file is a part of libmata.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef MATA_NFA_PLUMBING_HH_
+#define MATA_NFA_PLUMBING_HH_
+
+#include <mata/nfa.hh>
+
+namespace Mata {
+
+namespace Nfa {
+
+namespace Plumbing {
+
+    /// Make the transition relation complete.
+    inline void make_complete(
+            Nfa*             aut,
+            const Alphabet&  alphabet,
+            State            sink_state)
+    { // {{{
+        make_complete(*aut, alphabet, sink_state);
+    } // make_complete }}{
+
+    inline void complement(
+            Nfa*               result,
+            const Nfa&         aut,
+            const Alphabet&    alphabet,
+            const StringMap&  params = {{"algo", "classical"}},
+            std::unordered_map<StateSet, State> *subset_map = nullptr)
+    { // {{{
+        *result = complement(aut, alphabet, params, subset_map);
+    } // complement }}}
+
+    inline void minimize(Nfa* res, const Nfa &aut)
+    { // {{{
+        *res = minimize(aut);
+    } // minimize }}}
+
+    inline void determinize(
+            Nfa*        result,
+            const Nfa&  aut,
+            std::unordered_map<StateSet, State> *subset_map = nullptr)
+    { // {{{
+        *result = determinize(aut, subset_map);
+    } // determinize }}}
+
+    inline void reduce(
+            Nfa* result,
+            const Nfa &aut,
+            StateToStateMap *state_map = nullptr,
+            const StringMap&  params = {{"algorithm", "simulation"}})
+    { // {{{
+        *result = reduce(aut, state_map, params);
+    } // reduce }}}
+
+    inline bool is_universal(
+            const Nfa&         aut,
+            const Alphabet&    alphabet,
+            const StringMap&  params)
+    { // {{{
+        return is_universal(aut, alphabet, nullptr, params);
+    } // }}}
+
+
+    /**
+     * @brief Checks inclusion of languages of two NFAs: @p smaller and @p bigger (smaller <= bigger).
+     * @param params[in] Optional parameters to control the equivalence check algorithm:
+     * - "algo": "naive", "antichains" (Default: "antichains")
+     */
+    inline bool is_incl(
```

Comment:
wern't these interfaces meant to be in nfa.h instead of in nfa-plumbing.h?

Reply:
you're right, my fault, I've just moved it back. Fixed.